feature_engine/feature_selection.py

```diff
@@ -0,0 +1,69 @@
+# Authors: Soledad Galli <solegalli@protonmail.com>
+# License: BSD 3 clause
+
+import numpy as np
+import pandas as pd
+import warnings
+
+from sklearn.base import BaseEstimator, TransformerMixin
+from feature_engine.dataframe_checks import _is_dataframe
+from feature_engine.variable_manipulation import _define_variables
+
+
+class FeatureEliminator(BaseEstimator, TransformerMixin):
+    """
+    The FeatureEliminator() drops the list of variable(s) as provided by the user
+    from the dataframe and returns the subset of original dataframe with remaining
+    variables.
+
+    Parameters
+    ----------
+
+    features_to_drop : str or list, default=None
+        Desired variable/s to be dropped from the dataframe
+
+    """
+
+    def __init__(self, features_to_drop=None):
+
+        if not any(isinstance(features_to_drop, t) for t in [str, list]):
+            raise ValueError("features_to_drop must be a string or list object")
+
+        self.features = _define_variables(features_to_drop)
+
+    def fit(self, X, y=None):
+        """
+        Verifies that the passed input X if of the type pandas dataframe
+
+        Parameters
+        ----------
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+            The input dataframe on which the feature elimination has to be performed
+        y: None
+            y is not needed for this transformer
+
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
```

Comment:
the fit method needs to return at least 1 attribute with a trailing underscore, for example self.input_shape_, because the trailing underscore is that the is_fitted method checks when executing the transform method, to check that the user used fit to fit the transformer before using transform. I will point to example code as I go on with the review.

Reply:
fit method should include the `self.input_check`, as here:
https://github.com/solegalli/feature_engine/blob/master/feature_engine/variable_transformers.py#L222

I think we should also include a check to corroborate that the variables entered by the user are present in the dataframe. If not, return an error. 

What I am not sure is if we should also check absence of na in the training set. Thoughts?

Reply:
> the fit method needs to return at least 1 attribute with a trailing underscore, for example self.input_shape_, because the trailing underscore is that the is_fitted method checks when executing the transform method, to check that the user used fit to fit the transformer before using transform. I will point to example code as I go on with the review.

I will make sure that is added.

Reply:
> fit method should include the `self.input_check`, as here:
> https://github.com/solegalli/feature_engine/blob/master/feature_engine/variable_transformers.py#L222
> 
> I think we should also include a check to corroborate that the variables entered by the user are present in the dataframe. If not, return an error.
> 
> What I am not sure is if we should also check absence of na in the training set. Thoughts?

As far having a check for variables that doesn't exist, pandas drop() does raise a "KeyError" if it finds a variable not present in the dataframe. I also have a test case to check this (test case 2). Will that be sufficient or do you need an explicit check?